### PR TITLE
duplicate across layers new function

### PR DESCRIPTION
**Pull request by gkorompi**
duplicate across layer function.

### CODE REVIEW

File: tests/legacy/system/test_edb_modeler.py [hunk 1]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
             net5_length += path.length
         assert net5_length == 0.026285623899038543
         edbapp.close_edb()
+
+
+
+    def test_duplicate(self):
+        import pyedb
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:330]:
```suggestion

```

Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:330]:
```suggestion
```

File: tests/legacy/system/test_edb_modeler.py [hunk 2]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        import os
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:331]:
```suggestion
```

File: tests/legacy/system/test_edb_modeler.py [hunk 3]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        from pyedb.dotnet.edb import Edb
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:332]:
```suggestion
```

File: tests/legacy/system/test_edb_modeler.py [hunk 4]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        edbapp = Edb()
+        edbapp["$H"] = "0.65mil"
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:334]:
```suggestion
        edbapp["$H"] = "0.65mil"
        assert  edbapp["$H"] == "0.65mil"
```

File: tests/legacy/system/test_edb_modeler.py [hunk 5]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        edbapp.stackup.add_layer("bot_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d1", layer_type="dielectric", thickness="$S_D", material="FR4_epoxy")
+        edbapp.stackup.add_layer("trace2", thickness="$H")
+        edbapp.stackup.add_layer("d2", layer_type="dielectric", thickness="$T-$S_D", material="FR4_epoxy")
+        edbapp.stackup.add_layer("mid_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d3", layer_type="dielectric", thickness="13mil", material="FR4_epoxy")
+        edbapp.stackup.add_layer("top_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d4", layer_type="dielectric", thickness="13mil", material="FR4_epoxy")
+        edbapp.stackup.add_layer("trace1", thickness="$H")
+        r1 = edbapp.modeler.create_rectangle(center_point=("0,0"), width="200mil", height="200mil", layer_name="top_gnd", representation_type="CenterWidthHeight", net_name="r1")
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:348]:
```suggestion
        r1 = edbapp.modeler.create_rectangle(center_point=("0,0"), width="200mil", height="200mil", layer_name="top_gnd", representation_type="CenterWidthHeight", net_name="r1")
        assert r1
```

File: tests/legacy/system/test_edb_modeler.py [hunk 6]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        r2 = edbapp.modeler.create_rectangle(center_point=("0,0"), width="40mil", height="$Antipad_R*2", layer_name="top_gnd", representation_type="CenterWidthHeight", net_name="r2")
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:349]:
```suggestion
        r2 = edbapp.modeler.create_rectangle(center_point=("0,0"), width="40mil", height="$Antipad_R*2", layer_name="top_gnd", representation_type="CenterWidthHeight", net_name="r2")
        assert r2
```

File: tests/legacy/system/test_edb_modeler.py [hunk 7]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        c1 = edbapp.modeler.create_circle(layer_name="top_gnd", x="Via_S/2", y="0mil", radius="$Antipad_R", net_name="c1")
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:350]:
```suggestion
        c1 = edbapp.modeler.create_circle(layer_name="top_gnd", x="Via_S/2", y="0mil", radius="$Antipad_R", net_name="c1")
        assert c1
```

File: tests/legacy/system/test_edb_modeler.py [hunk 8]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        c2 = edbapp.modeler.create_circle(layer_name="top_gnd", x="-Via_S/2", y="0mil", radius="$Antipad_R", net_name="c2")
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:351]:
```suggestion
        c2 = edbapp.modeler.create_circle(layer_name="top_gnd", x="-Via_S/2", y="0mil", radius="$Antipad_R", net_name="c2")
        assert c2
```

File: tests/legacy/system/test_edb_modeler.py [hunk 9]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        r = [r2, c1, c2]
+        r1.subtract(r)
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:353]:
```suggestion
        assert r1.subtract(r)
```

File: tests/legacy/system/test_edb_modeler.py [hunk 10]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        lay_list = ["bot_gnd", "mid_gnd"]
+        edbapp.modeler.primitives_by_layer
+        edbapp.modeler.primitives[0].duplicate_across_layers(lay_list)
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:356]:
```suggestion
        assert edbapp.modeler.primitives[0].duplicate_across_layers(lay_list)
```

File: tests/legacy/system/test_edb_modeler.py [hunk 11]
```diff
@@ -323,3 +323,36 @@ def test_156_check_path_length(self):
...
+        edbapp.modeler.primitives_by_layer
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:357]:
```suggestion
        primitives = edbapp.modeler.primitives_by_layer
        assert len(primitives) == 2
```

Comment (svandenb-dev) [reply tests/legacy/system/test_edb_modeler.py:357]: Adjust the length I am not sure exactly how much we have

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Review by SMoraisAnsys — APPROVED** [review]

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Pull request merged**